src/sqlancer/clickhouse/gen/ClickHouseExpressionGenerator.java

```diff
@@ -53,48 +53,128 @@ private enum Expression {
         UNARY_POSTFIX
     }
 
-    public ClickHouseExpression generateExpressionWithColumns(List<ClickHouseColumnReference> columns, int depth) {
-        if (columns.isEmpty() || depth > 0 && Randomly.getBoolean()) {
+    public ClickHouseExpression generateExpressionWithColumns(List<ClickHouseColumnReference> columns, int complexity) {
```

Comment:
I think `complexity` is quite ambiguous. Perhaps `remainingDepth` would be a more descriptive term?

Reply:
Ok